qt/python/mantidqtinterfaces/mantidqtinterfaces/dns_powder_tof/script_generator/tof_powder_script_generator_model.py

```diff
@@ -0,0 +1,333 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+"""
+DNS script generator for TOF powder data
+"""
+
+from mantidqtinterfaces.dns_powder_tof.data_structures.dns_tof_powder_dataset \
+    import DNSTofDataset
+from mantidqtinterfaces.dns_powder_tof.script_generator.common_script_generator_model \
+    import DNSScriptGeneratorModel
+
+
+class DNSTofPowderScriptGeneratorModel(DNSScriptGeneratorModel):
```

Comment:
I can see how much you have done on this p[rocess, but how much of it could be handled running the functions and using the workspace fistory?

Reply:
I see your point. I am not sure why it was decided to not use the workspace history for the script generation purposes since I joined at the final stages of the GUI's development

Reply:
Do you think it would be possible to change it to utalize workspace history before the code freeze?

Reply:
Would it be OK if we leave this to the subsequent PR which will contain another operational mode of the GUI? That mode also contains the script generator model, so it will need to be updated anyways. I am attending a workshop/school this week and the schedule is dense, so I don't have an opportunity to implement it this week

Reply:
That will be acceptable.